Iris-Pipeline/steps/validate.py

```diff
@@ -0,0 +1,196 @@
+        X=X_validate,
+        y=y_validate,
+        )
+    
+    for count, (score) in enumerate(scores):
+        mlflow.log_metric(
+            f"val_score", score
+        )
+
+
+def save_model(
+    dtc_model: DecisionTreeClassifier,
+    model_name: str,
+    models_path: str
+    ):        
+    mlflow.sklearn.save_model(
+        sk_model=dtc_model,
+        path=models_path,
+        serialization_format=mlflow.sklearn.SERIALIZATION_FORMAT_PICKLE,
+    )
+
+
+@click.command()
+@click.option("--process_run_id")
+@click.option("--train_run_id")
+@click.option("--config_path")
+def task(process_run_id, train_run_id, config_path):
+    """
+    Validate model performance against deployed model
+
+    Orchestrates the validation of model performance. 
+    This is done by first evaluating the model using this function:
+        evaluate(): Evaluates the model using the validation dataset
+    Then it compares the model to the one that is currently deployed using this function:
+        compare_models(): Compares the deployed model metrics to the current model metrics
+    
+    Args:
+        process_run_id (str): run id that the process step generated
+        train_run_id (str): run id that the train step generated
+        config_path (str): path to the configuration file
+    """
+    with mlflow.start_run() as mlrun:
+        with open(config_path, "r", encoding="UTF-8") as ymlfile:
+            cfg = yaml.load(ymlfile, Loader=yaml.FullLoader)
+            
+        json_path = Path("../",cfg["model_path"], cfg["deployment_json_name"])
+        with open(json_path, 'r') as openfile:
+            json_object = json.load(openfile)
+        
+        model_version = train_run_id[:5]    
+        model_name = cfg["model_name"] + f"-{model_version}"
+         
+        process_run = mlflow.tracking.MlflowClient().get_run(process_run_id)
+        
+        validate_path = Path(process_run.info.artifact_uri, cfg["validate_data_name"])
+        validate_df = pd.read_csv(validate_path)
+        
+        model_path = f"runs:/{train_run_id}/{model_name}"
+        deployed_path = Path("../",cfg["model_path"],json_object['name'])
+        
+        dtc_model = mlflow.sklearn.load_model(str(model_path))
+        
+        X_validate = validate_df[cfg["features"].keys()]
+        y_validate = validate_df[cfg["label_name"]]
+        
+        evaluate(
+            model=dtc_model,
+            X_validate=X_validate,
+            y_validate=y_validate
+            )
+        
+        if deployed_path.exists():
+            deployed_model = mlflow.sklearn.load_model(str(deployed_path))
+            compare_models(
+                deployed_model=deployed_model,
+                current_model=dtc_model,
+                X_validate=X_validate,
+                y_validate=y_validate,
+                json_path=json_path,
+                model_name=model_name
+                )
+        else:
+            with open(json_path, 'r') as openfile:
```

Comment:
I found that this is what stackoverflow recommends you do to change what the contents of a json file is, because just reassigning to it kept giving me problems